### PR TITLE
Update testing to new approach fixes #31

### LINKED ISSUE #31: Update testing to new approach
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### PR DESCRIPTION
**Pull request by ingwarr**
Fixes https://github.com/terraform-google-modules/terraform-google-vm/issues/31

### CODE REVIEW

File: examples/compute_instance/simple/README.md [hunk 1]
```diff
@@ -22,4 +22,4 @@ This is a simple, minimal example of how to use the compute_instance module
 | available\_zones | List of available zones in region |
 | instances\_self\_links | List of self-links for compute instances |
 
-[^]: (autogen_docs_end)
\ No newline at end of file
+[^]: (autogen_docs_end)
```
Comment (ivankorn) [inline examples/compute_instance/simple/README.md]: this is the old style doc generator marker, please replace it across the code base using [this function](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/concourse/build/developer-tools/build/scripts/task_helper_functions.sh#L186)

File: examples/mig_with_percent/simple/README.md [hunk 2]
```diff
@@ -21,4 +21,4 @@ managed instance group.
 |------|-------------|
 | self\_link | Self-link of the managed instance group |
 
-[^]: (autogen_docs_end)
\ No newline at end of file
+[^]: (autogen_docs_end)
```
Comment (ivankorn) [inline examples/mig_with_percent/simple/README.md]: this is the old style doc generator marker, please replace it across the code base using [this function](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/concourse/build/developer-tools/build/scripts/task_helper_functions.sh#L186)

File: modules/compute_instance/README.md [hunk 3]
```diff
@@ -31,4 +31,4 @@ See the [simple](examples/compute_instance/simple) for a usage example.
 | available\_zones | List of available zones in region |
 | instances\_self\_links | List of self-links for compute instances |
 
-[^]: (autogen_docs_end)
\ No newline at end of file
+[^]: (autogen_docs_end)
```
Comment (ivankorn) [inline modules/compute_instance/README.md]: this is the old style doc generator marker, please replace it across the code base using [this function](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/concourse/build/developer-tools/build/scripts/task_helper_functions.sh#L186)

Comment (ingwarr) [reply modules/compute_instance/README.md]: Fixed

File: examples/mig_with_percent/simple/README.md (hunk 2 above)
Comment (ingwarr) [reply examples/mig_with_percent/simple/README.md]: fixed

File: examples/compute_instance/simple/README.md (hunk 1 above)
Comment (ingwarr) [reply examples/compute_instance/simple/README.md]: fixed

File: examples/compute_instance/simple/README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 This is a simple, minimal example of how to use the compute_instance module
 
-[^]: (autogen_docs_start)
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (Jberlinsky) [inline examples/compute_instance/simple/README.md]: Please run `make docker_generate_docs`.

File: examples/mig_with_percent/simple/README.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 This is a simple, minimal example of how to use the MIG with percent module to create a
 managed instance group.
 
-[^]: (autogen_docs_start)
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (Jberlinsky) [inline examples/mig_with_percent/simple/README.md]: Please run `make docker_generate_docs`.

File: modules/compute_instance/README.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ See the [simple](examples/compute_instance/simple) for a usage example.
 ## Testing
 
 
-[^]: (autogen_docs_start)
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (Jberlinsky) [inline modules/compute_instance/README.md]: Please run `make docker_generate_docs`.

File: Makefile [hunk 7]
```diff
@@ -18,133 +18,68 @@
...
 .PHONY: docker_run
 docker_run:
 	docker run --rm -it \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && exec /bin/bash"
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile [hunk 8]
```diff
@@ -18,133 +18,68 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && kitchen converge"
-
-.PHONY: docker_verify
-docker_verify:
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile [hunk 9]
```diff
@@ -18,133 +18,68 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && kitchen create"
-
-.PHONY: docker_converge
-docker_converge:
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile [hunk 10]
```diff
@@ -18,133 +18,68 @@
...
+.PHONY: docker_test_integration
+docker_test_integration:
 	docker run --rm -it \
-		-e PROJECT_ID \
 		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && kitchen verify"
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile [hunk 11]
```diff
@@ -18,133 +18,68 @@
...
+.PHONY: docker_test_lint
+docker_test_lint:
 	docker run --rm -it \
-		-e PROJECT_ID \
-		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && kitchen destroy"
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: Makefile [hunk 12]
```diff
@@ -18,133 +18,68 @@
...
+.PHONY: docker_generate_docs
+docker_generate_docs:
 	docker run --rm -it \
-		-e PROJECT_ID \
-		-e SERVICE_ACCOUNT_JSON \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "cd /cft/workdir && source test/ci_integration.sh && setup_environment && make test_integration"
+		-v $(CURDIR):/workspace \
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply Makefile]: Fixed

File: examples/compute_instance/simple/README.md (hunk 4 above)
Comment (ingwarr) [reply examples/compute_instance/simple/README.md]: Fixed

File: examples/mig_with_percent/simple/README.md (hunk 5 above)
Comment (ingwarr) [reply examples/mig_with_percent/simple/README.md]: Fixed

File: modules/compute_instance/README.md (hunk 6 above)
Comment (ingwarr) [reply modules/compute_instance/README.md]: Fixed

**Pull request merged**